hoomd/hpmc/ExternalFieldWall.h

```diff
@@ -629,6 +629,56 @@ template<class Shape> class ExternalFieldWall : public ExternalFieldMono<Shape>
         return double(0.0);
         }
 
+    //! Evaluate the energy of the force.
+    /*! \param box The system box.
+        \param type Particle type.
+        \param r_i Particle position
+        \param q_i Particle orientation.
+        \param diameter Particle diameter.
+        \param charge Particle charge.
+        \returns Energy due to the force
+    */
+    virtual float energy(const BoxDim& box,
```

Comment:
```suggestion
    virtual Scalar energy(const BoxDim& box,
```

Reply:
not sure what the benefit is of having this as a float or scalar but happy to commit if it makes sense.

Reply:
so when I implemented the changes from float back to Scalar, I started getting a bunch of errors while building. I think that replacing float with scalar might be more complicated than at first glance. Since this is a bug fix with code that's going to be rewritten soon anyway I think it's fine to leave as float for now.